### ISSUE TITLE
Travis CI crumbling, time for another option?

### ISSUE THREAD

**Issue by Yetangitu** [airsonic/airsonic#1238]
I'm seeing more and more random build failures which are totally unrelated to submitted changes. By now it is more or less expected for Travis to complain and actually a surprise when it doesn't. Give it a kick and Travis might pop up with another random build failure, if in luck it succeeds. This problem is not limited to _airsonic_, I see it everywhere Travis is used. As to whether this is related to the fact that the company was [bought by Idera](https://news.ycombinator.com/item?id=18978251) about half a year ago I do not know but the frequency of these random failures has made Travis more of a hinder than an asset by now.

I'd say it is time to switch to another CI provider, that HN discussion linked to above points out a few possible candidates:

Circle CI: https://circleci.com/

Buildkite: https://buildkite.com/

Azure Pipelines: https://azure.microsoft.com/en-us/services/devops/pipelines
 - https://docs.microsoft.com/azure/devops/pipelines/migrate/from-travis

Github Actions: https://github.com/features/actions

As far as I know all of these offer free tiers for free software projects.

**Comment by jvoisin** [comment]
Don't forget (the amazing) [gitlab ci]( https://gitlab.com/airsonic/airsonic )

**Comment by tesshucom** [comment]
I learned by #1240 that  Airsonic is trying to run integration tests in parallel.
But seems  to be difficult.
integration tests have many barrier factors to parallelization.
(LFS or flash IO of DB and Lucene etc)

Other CI with very fast IO may work it temporarily, but after a while the same result will be achieved.
(Because of transition competition between users.)

Execution of parallel processing by software is not so versatile.

**Comment by eharris** [comment]
I'm hoping that #1242 will help fix the Travis CI reliability issues by switching from parallel threaded to single forked mode.

**Comment by Yetangitu** [comment]
> Don't forget (the amazing) [gitlab ci](https://gitlab.com/airsonic/airsonic)

Sure, and here I thought that'd imply a move to Gitlab (fine by me but that is another question for another day) and was proven wrong: 

https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html

**Comment by jvoisin** [comment]
We already have a gitlab mirror anyway :)

**Comment by eharris** [comment]
The change to not do tests in parallel (#1242) seems to have fixed the Travis flakiness, so I'm going to close this as resolved.  If someone wants to pick up moving off Travis to something else, this can either be reopened or a new issue can be created with the specific plan.
